### PR TITLE
feat: secilc for building SELinux policies

### PR DESCRIPTION
**Pull request by dsseng**
Tested, works in the toolchain container in docker for successfully building my WIP policy

We might be interested in building CIL in runtime like Android does that, but that would be required if we will need to bundle policy at runtime

### CODE REVIEW

**Comment by dsseng** [comment]
Will sign tomorrow, having a bit of remote-access related trouble, will sign on local perhaps

**Review by smira — APPROVED** [review]

File: Pkgfile [hunk 1]
```diff
@@ -299,6 +304,11 @@ vars:
   rhash_sha256: 8e7d1a8ccac0143c8fe9b68ebac67d485df119ea17a613f4038cda52f84ef52a
   rhash_sha512: 00a7e5e058b53ce20ae79509815452ed9cb699d1322b678220b72c61dea3ea2f8fa131acfade8bb6d9f6af913f0c3c472330841181b22314b8755166310c946f
 
+  # renovate: datasource=github-tags extractVersion=^(?<version>.*)$ depName=SELinuxProject/selinux
```
Comment (frezbo) [inline Pkgfile:307]: we don't need the `extractVersion` though this needs to be added to https://github.com/siderolabs/tools/blob/main/.github/renovate.json#L57

File: secilc/patches/disable-manpages.patch [hunk 2]
```diff
@@ -0,0 +1,30 @@
+diff --git a/Makefile b/Makefile
```
Comment (frezbo) [inline secilc/patches/disable-manpages.patch:1]: we can just `rm -rf` on the man folder rather than including a patch. or does it not work at all?

**Review by frezbo — APPROVED** [review]

File: tools/pkg.yaml [hunk 3]
```diff
@@ -40,6 +40,7 @@ dependencies:
   - stage: libcap
   - stage: libffi
   - stage: libnl
+  - stage: libsepol
```
[the remark this answers is not in the document]
Comment (dsseng) [reply tools/pkg.yaml:43]: Isn't tools the reproducibility target itself? Could not find any special `reproducibility` package like in pkgs

File: secilc/patches/disable-manpages.patch (hunk 2 above)
Comment (dsseng) [reply secilc/patches/disable-manpages.patch:1]: Ah, we needed xml2man for that, so I decided to patch out building that altogether

File: Pkgfile (hunk 1 above)
Comment (dsseng) [reply Pkgfile]: @frezbo did I do that right? /m if so

File: secilc/patches/disable-manpages.patch (hunk 2 above)
Comment (frezbo) [reply secilc/patches/disable-manpages.patch:1]: ah okay,  guess we have to live with this patch, wished this was configure based

File: Pkgfile [hunk 4]
```diff
@@ -178,6 +178,11 @@ vars:
   libnl_sha256: 9fe43ccbeeea72c653bdcf8c93332583135cda46a79507bfd0a483bb57f65939
   libnl_sha512: 80fbbc079299c90afd2a5eda62e4d4f98bf4ef23958c3ce5101f4ed4d81d783af733213bb3bab15f218555d8460bc2394898f909f4ac024fc27281faec86a041
 
+  # renovate: datasource=github-tags depName=SELinuxProject/selinux
```
Comment (frezbo) [inline Pkgfile:181]:
```suggestion
  # renovate: datasource=git-tags extractVersion=^libsepol-(?<version>.*)$  depName=https://github.com/SELinuxProject/selinux.git
```

File: Pkgfile [hunk 5]
```diff
@@ -299,6 +304,11 @@ vars:
   rhash_sha256: 8e7d1a8ccac0143c8fe9b68ebac67d485df119ea17a613f4038cda52f84ef52a
   rhash_sha512: 00a7e5e058b53ce20ae79509815452ed9cb699d1322b678220b72c61dea3ea2f8fa131acfade8bb6d9f6af913f0c3c472330841181b22314b8755166310c946f
 
+  # renovate: datasource=github-tags depName=SELinuxProject/selinux
```
Comment (frezbo) [inline Pkgfile:307]:
```suggestion
  # renovate: datasource=git-tags extractVersion=^secilc-(?<version>.*)$ depName=https://github.com/SELinuxProject/selinux.git
```

File: .github/renovate.json [hunk 6]
```diff
@@ -37,6 +37,7 @@
                 "mirror/ncurses",
                 "PCRE2Project/pcre2",
                 "plougher/squashfs-tools",
+                "SELinuxProject/selinux",
```
Comment (frezbo) [inline .github/renovate.json:40]: let's use `https://github.com/SELinuxProject/selinux.git` here

Comment (dsseng) [reply .github/renovate.json:40]: ok, currently on the tcl stuff, will fix and then return here (if not busy with other stuff)

File: Pkgfile (hunk 4 above)
Comment (dsseng) [reply Pkgfile:181]: Tags like 3.7 plain should contain everything: https://github.com/SELinuxProject/selinux/releases/tag/3.7

Comment (frezbo) [reply Pkgfile]: ahh then it's fine, let's keep renovate config json as is then https://github.com/siderolabs/tools/pull/379/files#diff-f82f7d36a61d22f640c25bdb8b0435bf9cb38679d99d5f460753fa668ee7cdb3R40 here

**Review by frezbo — APPROVED** [review]

**Comment by dsseng** [comment]
Works for building our sources

/m

**Pull request merged**